src/toil/wdl/wdltoil.py

```diff
@@ -911,14 +995,24 @@ def devirtualize_to(filename: str, dest_dir: str, file_source: Union[AbstractFil
                 virtualized_to_devirtualized[filename] = result
             logger.debug('Devirtualized %s as openable file %s', filename, result)
         else:
-            # This is a local file
-            # To support relative paths, join the execution dir and filename
-            # if filename is already an abs path, join() will do nothing
-            if execution_dir is not None:
-                result = os.path.join(execution_dir, filename)
-            else:
-                result = filename
-            logger.debug("Virtualized file %s is already a local path", filename)
+            result = None
+            if wdl_options.get("filename_to_uri") is not None:
+                # We support importing files relative to a URL when the input JSON itself is downloaded
+                # See import_url_files()
+                filename_uri = wdl_options["filename_to_uri"].get(filename)
+                if filename_uri is not None:
+                    # This was previously virtualized, so use the virtualized result
+                    result = ToilWDLStdLibBase._devirtualize_uri(filename_uri, dest_dir, file_source)
```

Comment:
I'm not sure this comment describes what is happening here.

Reply:
Removed